src/menu/views/load_rom.c

```diff
@@ -359,8 +368,19 @@ void view_load_rom_init (menu_t *menu) {
             path_free(menu->load.rom_path);
         }
 
-        menu->load.rom_path = path_clone_push(menu->browser.directory, menu->browser.entry->name);
-    }
+        if(menu->load.load_history != -1) {
+            menu->load.rom_path = path_clone(menu->bookkeeping.history_items[menu->load.load_history].primary_path);
+        } else if(menu->load.load_favorite != -1) {
+            menu->load.rom_path = path_clone(menu->bookkeeping.favorite_items[menu->load.load_history].primary_path);
+        } else {
+            menu->load.rom_path = path_clone_push(menu->browser.directory, menu->browser.entry->name);
+        }
+
+        name = path_last_get(menu->load.rom_path);
+    }    
+
+    menu->load.load_favorite = -1;
```

Comment:
something here in my loins says for readability/maintainability it should be something like `menu->bookkeeping.get_favorite = -1;`

and probably the whole function above...

Reply:
I did have these in the bookkeeping struct before, but figured it might be better in the load with the other variables.
but fine with where ever.